keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1215 @@
+block traffic with malicious in-cluster clients, or workloads that pose security risks.
+Those traffic restrictions can only be lifted once the `Deny` rules are deleted, 
+modified by the admin, or overridden by a higher priority rule.
+
+On the other hand, the `Allow` rules can be used to call out traffic in the cluster
+that needs to be allowed for certain components to work as expected (egress to
+CoreDNS for example). Those traffic should not be blocked when developers apply
+NetworkPolicy to their Namespaces which isolates the workloads.
+
+AdminNetworkPolicy `Pass` rules allows an admin to delegate security posture for
+certain traffic to the Namespace owners by overriding any lower priority Allow or 
+Deny rules. For example, allowing intra-tenant traffic can be delegated to tenant 
+admins explicitly by the cluster admin with the use of `Pass` rules.
+
+#### Priority
+
+The policy instances will be ordered based on the numeric priority assigned to the
+ANP. `Priority` is a 32 bit integer value, where a larger number corresponds to 
+a higher precedence. The lowest "regular" value (see below for 
+special cases) is "1", which corresponds to the lowest "importance". High values 
+have higher importance. For alpha, this API defines "1000" as the maximum value, 
+but this may be revisited as the proposal advances. For future-safety, clients may 
+assume that higher values will eventually be allowed, and simply treat it as an int32.
+Any positive priority will have higher precedence over the namespaced NetworkPolicy 
+instances in the cluster, unless the traffic matches on a `Pass` rule that allows 
+it to bypass any lower positive priority ANP rules.
+
+Additionally, the special priority "0" can be used in the priority field to indicate
+that the rules in that policy instance shall be created at a priority lower than the
+Namespaced NetworkPolicies. 
+
+The relative precedence of the rules within a single ANP object (all of which 
+share a priority) will be determined by the order in which the rule is written. 
+Thus, a rule that appears at the top of the ingress/egress rules would take the 
+highest precedence. The maximum number of rules, which will be calculated as the 
+total summation of the AdminNetworkPolicyIngressRules and AdminNetworkPolicyEgressRules
+in a single ANP instance, will be 100.
+
+Conflict resolution: Two policies are considered to be conflicting if they are assigned the
+same `priority` and apply to the same resources or a union of resources. In order to avoid such conflicts, 
+we propose to include tooling for ANP resources to help alert the admin to potentially ambiguous ANP
+priority scenarios, more details in [risks and mitigation](#risks-and-mitigation). However, ultimately
+it will be the job of the network policy implementation to decide how to handle overlapping priority 
+situations. 
+
+#### Rule Names
+
+In order to help future proof the ANP api, a built in mechanism to identify each
```

Comment:
How does this help future-proof? I can't immediately invent a scenario, but that doesn't mean it's a bad idea.

Reply:
Say for example we decide to incorporate per rule logging for ANP..... rule naming would make the logging much simpler to understand. 

Also for the tooling we plan to implement,  we would like to be able to alert the Cluster Admin as to wether or not a NP would be overridden, this feature would allow the tool to report exactly what rule/peer is causing the issue

Reply:
But every rule still has a unique name. It's just the name of the containing object + the name of the Rule.

Reply:
Or the name of the object plus the index of the rule (although that changes over time, does it need to be unique over all of history or just at a given moment in time?)